tests/test_device.py

```diff
@@ -722,6 +724,48 @@ def test_outdated_API(self, monkeypatch):
             with pytest.raises(DeviceError, match="plugin requires PennyLane versions"):
                 qml.device("default.qubit", wires=0)
 
+    def test_refresh_entrypoints(self, monkeypatch):
+        """Test that new entrypoints are found by the refresh_devices function"""
+
+        with monkeypatch.context() as m:
+            # remove all entry points
+            m.setattr(pkg_resources, "iter_entry_points", lambda name: [])
+
+            importlib.reload(qml)
+
+            # since there are no entry points, there will be no plugin devices
+            assert not qml.plugin_devices
+
+        # outside of the context, entrypoints will now be found
+        assert not qml.plugin_devices
+        qml.refresh_devices()
+        assert qml.plugin_devices
+
+        importlib.reload(qml)
```

Comment:
Maybe worth adding a comment why this call is necessary.

Reply:
Done!